setup.py

```diff
@@ -1,14 +1,32 @@
 from __future__ import absolute_import, division, print_function
 
+import subprocess
 import sys
 
+import os
 from setuptools import find_packages, setup
 
+PROJECT_PATH = os.path.dirname(os.path.abspath(__file__))
+VERSION = """
+# This file is auto-generated with the version information during setup.py installation.
+
+__version__ = '{}'
+"""
+
 # Find pyro version.
-for line in open('pyro/__init__.py'):
-    if line.startswith('__version__ = '):
+for line in open(os.path.join(PROJECT_PATH, 'pyro', '__init__.py')):
+    if line.startswith('version_prefix = '):
         version = line.strip().split()[2][1:-1]
 
+# Append current commit sha to version
+commit_sha = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD'],
```

Comment:
Do you think it's work putting this in a `try-except` in case some use case isn't using a git clone?
```py
try:
    comit_sha = subprocess.check_output(...).strip()
    version += '+{}'.format(commit_sha)
except OSError:
    pass
```

Reply:
Yes, absolutely.